manifests/ovs-cni.yml.in

```diff
@@ -36,7 +37,78 @@ spec:
         volumeMounts:
         - name: cnibin
           mountPath: /host/opt/cni/bin
+      - name: ovs-cni-marker
+        image: ${OVS_CNI_MARKER_IMAGE_REPO}/${OVS_CNI_MARKER_IMAGE_NAME}:${OVS_CNI_MARKER_IMAGE_VERSION}
+        imagePullPolicy: ${OVS_CNI_MARKER_IMAGE_PULL_POLICY}
+        securityContext:
+          privileged: true
+        command:
+          - sh
+          - -c
+          - |
+            if [ -e /host/var/run/openvswitch/db.sock ]; then
+              echo "openvswitch running on host"
+              ovs_socket="unix:///host/var/run/openvswitch/db.sock"
+            elif [ -e /host/run/openvswitch/db.sock ]; then
+              echo "openvswitch taken from openshift"
+              ovs_socket="unix:///host/run/openvswitch/db.sock"
+            else
+              exit 1
+            fi
+            /marker -v 3 -logtostderr -node-name $(NODE_NAME) -poll-interval 10 -ovs-socket ${ovs_socket}
+        volumeMounts:
+          - name: ovs-var-run
+            mountPath: /host/var/run/openvswitch
+          - name: ovs-run
+            mountPath: /host/run/openvswitch
+        env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
       volumes:
         - name: cnibin
           hostPath:
             path: /opt/cni/bin
+        # openvswitch running on host
```

Comment:
This seems too liberal to expose both paths just in case. I suggest we have two separate manifests per platform. What do you think?

Reply:
Sure, If we don't mind it being duplicated.

Then we can have kubernetes-ovs-cni deploying ovs-cni AND openshift-ovs-cni that would cover both ovs and ovs-cni, then we have the path under control. What do you think?

Reply:
Sorry, I perhaps misworded above saying 'two separate manifests per platform', while I meant 'two separate manifests, one per platform'.

I don't follow your suggestion. Why would we deploy openshift-ovs-cni manifest in addition to ovs-cni for non-openshift clusters? (And vice versa.) Perhaps I am not reading your plan right. Could you please elaborate?

Reply:
**kubernetes-ovs-cni.yml** will deploy ovs-cni daemon set.

**openshift-ovs-cni.yml** will deploy ovs-cni daemon set and ovs-vsctl (deprecates openshift-ovs-vsctl.yml).

Just that.

Reply:
OK then I am a master of misreading. Where's my medal.

Yes this makes sense.